### DISCUSSION TITLE
Theme creator

### DISCUSSION THREAD

**Discussion by Alonely0** [GitSquared/edex-ui#1028] [Ideas]
You could add a feature for creating a theme by clicking objects and setting color, moving elements to other positions and then inject the custom CSS, etc. Create a theme for eDEX is very difficult if you don't know how JSON works or if you don't have CSS knowledge (that's my case).

**Comment by wrac4242** [comment]
possibly can be done by having a template CSS code, and an ability to modify that, and refresh the display colour/theme, however i am not sure how that would work on the display

**Comment by GitSquared** [comment]
(converted to Discussion, as this is not a technical bug)

I don't think making a WYSIWYG editor would be the right solution, I think more/better customization options in eDEX should require some basic hacking skills (and make apprentice tinkerers want to learn them).

The problem with the current solution is that it was created as an afterthought, because of how cheap it was to implement and how many possibilities it opened, and JSON/CSS editing simply isn't fun. It's boring and quirky and it breaks all the time when you're trying to override an application's styling.

I think as a whole if I got the chance to revive this project it would be much more focused on customization, hacking the software to make it work & look like you want, and giving you the motivation to start learning if you haven't already. Maybe one day...

**Reply by Alonely0** [reply]
Agree, but there is the point, we need to revive it. I think that the things to implement yes or yes would have to be:
  * Don't bloat it by default, let decide users what want of edex ui to run and what not
  * Full appearance customization without having to override manually the CSS
  * Implement user-defined panels
  * Some integration with the shell (specially ZSH)
  * Make edex faster by using GPU like in alacritty and kitty (let the user decide)
  * Make edex be usable without being maximized or in full screen, because usually goes brr when using a tiling window manager (just remove all panels when the app is not maximized or in full screen)

**Reply by GitSquared** [reply]
Agree with most/all of those points, but doing this correctly would require a full rewrite, which takes time (and time is money).

**Reply by Alonely0** [reply]
Agree with that. We could make something like edex-reborn this summer.
